src/Symfony/Component/Routing/StaticSiteGeneration/StaticPageUrisProvider.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Routing\StaticSiteGeneration;
+
+use Psr\Container\ContainerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Exception\InvalidArgumentException;
+use Symfony\Component\Routing\Exception\LogicException;
+use Symfony\Component\Routing\RouterInterface;
+
+/**
+ * @author Thomas Bibaut <bibaut.t@gmail.com>
+ */
+final readonly class StaticPageUrisProvider implements StaticPageUrisProviderInterface
+{
+    public function __construct(
+        private RouterInterface $router,
+        private ContainerInterface $paramsProviders,
+    ) {
+    }
+
+    /**
+     * WARNING: This method should never be used at runtime as it is SLOW.
+     *
+     * @return iterable<string>
+     *
+     * @throws LogicException           if a route noted static is not configured properly
+     * @throws InvalidArgumentException if some route parameters cannot be met
+     */
+    public function provide(): iterable
+    {
+        foreach ($this->router->getRouteCollection() as $routeName => $route) {
+            /** @var bool|array{params?: string|iterable<array<mixed>>} */
+            $config = $route->getDefaults()['_static_generation'] ?? null;
+            if (!$config) {
+                continue;
+            }
+
+            if ([] !== $route->getMethods() && !\in_array(Request::METHOD_GET, $route->getMethods(), true)) {
```

Comment:
all comparisons to bool/empty values should be simplified in the PR - those are needless boilerplate:

```suggestion
            if ($route->getMethods() && !\in_array(Request::METHOD_GET, $route->getMethods(), true)) {
```